apps/src/gamelab/AnimationTab/AnimationListItem.jsx

```diff
@@ -93,15 +94,26 @@ const AnimationListItem = React.createClass({
     style: React.PropTypes.object,
   },
 
+  getInitialState: function () {
+    return {
+      frameDelay: this.props.animationList.propsByKey[this.props.animationKey].frameDelay,
+      name: this.props.animationList.propsByKey[this.props.animationKey].name,
+      isNameValid: true
+    };
+  },
+
   componentWillReceiveProps(nextProps) {
     if (this.props.columnWidth !== nextProps.columnWidth) {
       this.refs.thumbnail.forceResize();
     }
-    this.setState({frameDelay: nextProps.animationProps.frameDelay});
+    this.setState({frameDelay: nextProps.animationList.propsByKey[this.props.animationKey].frameDelay});
```

Comment:
If you rewrite the `getAnimationProps()` suggestion above to take `props` as an argument you could use it to get the animation props from `nextProps` too.  That would also fix what _might_ be a subtle bug here - should we be using `nextProps.animationKey` instead of `this.props.animationKey` to index into `nextProps.animationList.propsByKey`?